### PR TITLE
Include layout transformation ops in extended minimal build and above.

### PR DESCRIPTION
**Pull request by edgchen1**
**Description**
Always include layout transformation ops in an extended minimal build or above.

**Motivation and Context**
It was previously possible to remove ops used by layout transformation in a reduced ops build. However, these are required in a minimal extended build or higher where layout transformation is enabled.

### CODE REVIEW

**Comment by justinchuby** [comment]
It'd be great if you could wait for https://github.com/microsoft/onnxruntime/pull/11318

File: onnxruntime/core/framework/kernel_def_hash_helpers.cc [hunk 1]
```diff
@@ -13,9 +13,10 @@ std::optional<HashValue> GetHashValueFromStaticKernelHashMap(const std::string&
   // Since layout transformation can happen in an extended build, if these nodes are not picked up and compiled by
   // NNAPI or other compiling EPs then we need a way to get the hashes for these nodes. Since the infrastructure
   // as well as op_schema required to generate these hashes is not available in an extended minimal build,
-  // we maintain a static map of nodes to hash value. This hash value can then be used to retireive the
+  // we maintain a static map of nodes to hash value. This hash value can then be used to retrieve the
   // kernel for the given op.
-  static std::unordered_map<std::string, HashValue> static_kernel_hashes{
+  static const std::unordered_map<std::string, HashValue> static_kernel_hashes{
+      // @@region_begin(layout_transformation_required_kernels)@@
       {"Transpose_1", 4324835766923221184ULL},
```
Comment (skottmckay) [inline onnxruntime/core/framework/kernel_def_hash_helpers.cc:20]: Should we mention what uses these carefully formatted markers?

**Comment by skottmckay** [comment]
>     '''

nit: add is_extended_minimal_build_or_higher to doc

---
Refers to: tools/ci_build/reduce_op_kernels.py:261 in fa4e1c2. [](commit_id = fa4e1c25bbf46bdea2ac50e0ed9ecee21e7c774c, deletion_comment = False)

File: tools/ci_build/reduce_op_kernels.py [hunk 2]
```diff
@@ -25,17 +27,81 @@
...
+                         required_optypes: typing.Set[str]):
+                self.filter_to_adapt = filter_to_adapt
+                self.required_optypes = required_optypes
+
+            def is_typed_registration_needed(self, domain: str, optype: str, type_registration_str: str):
+                # Always require registration for ONNX ops in self.required_optypes.
+                if domain == 'ai.onnx' and optype in self.required_optypes:
+                    return True
+                return self.filter_to_adapt.is_typed_registration_needed(domain, optype, type_registration_str)
+
```
Comment (skottmckay) [inline tools/ci_build/reduce_op_kernels.py:80]: side note: if needed in the future we could probably check the model for the small set of layout sensitive operators to determine the exact types required. I don't think it's worth doing currently though.

**Review by skottmckay — APPROVED** [review]
:shipit:

**Review by skottmckay — APPROVED** [review]
:shipit:

**Pull request merged**